### PR TITLE
Use dama/doctrine-test-bundle for proper test DB isolation

### PR DESCRIPTION
**Pull request by dmaicher**
This fixes https://github.com/symfony/symfony-demo/issues/445 by using https://github.com/dmaicher/doctrine-test-bundle to have proper test isolation.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -28,4 +28,8 @@
             </exclude>
         </whitelist>
     </filter>
+
+    <listeners>
+        <listener class="\DAMA\DoctrineTestBundle\PHPUnit\PHPUnitStaticDbConnectionListener" file="vendor/dama/doctrine-test-bundle/src/DAMA/DoctrineTestBundle/PHPUnit/PHPUnitStaticDbConnectionListener.php" />
```
Comment (javiereguiluz) [inline phpunit.xml.dist]: Is the `file` attribute required here? It looks a bit ugly 😢

Comment (dmaicher) [inline phpunit.xml.dist]: You are right :blush: It's not required :+1:

**Comment by dmaicher** [comment]
@javiereguiluz I added a few comments explaining the usage of this bundle. Should I also comment the listener inside `phpunuit.xml.dist`?

File: tests/AppBundle/Controller/Admin/BlogControllerTest.php [hunk 2]
```diff
@@ -80,44 +80,44 @@ public function testAdminBackendHomePage()
     }
 
     /**
-     * This test will manipulate the database state by deleting a Post.
-     * All changes to the database are rolled back before the next test case is executed.
-     * This is done by using https://github.com/dmaicher/doctrine-test-bundle.
-     *
-     * Therefore in the following test case we can edit the Post with ID=1 although we deleted it in this test.
+     * This test changes the database contents by deleting a blog post. However,
+     * thanks to the DAMADoctrineTestBundle and its PRPUnit listener, all changes
```
Comment (dmaicher) [inline tests/AppBundle/Controller/Admin/BlogControllerTest.php]: `PRPUnit`? :smile:

Comment (javiereguiluz) [inline tests/AppBundle/Controller/Admin/BlogControllerTest.php]: Fixed!

**Comment by dmaicher** [comment]
@javiereguiluz changes look good to me :+1:

**Comment by javiereguiluz** [comment]
@dmaicher I'm sorry but when merging other PR, a conflict was created in your composer.lock file. Could you please rebase it? Thanks!

**Comment by dmaicher** [comment]
Sure will do it tonight :wink:

**Comment by dmaicher** [comment]
@javiereguiluz done :wink:

**Comment by javiereguiluz** [comment]
@dmaicher thanks! I've merged this to test it for real in the application and see how it goes :)

**Comment by dmaicher** [comment]
:+1:

**Pull request merged**